diff -ruN a/src/lib.rs b/src/lib.rs
--- a/src/lib.rs	2026-08-16 16:57:01.492407443 +0900
+++ b/src/lib.rs	2026-08-16 16:57:01.506313648 +0900
@@ -36,6 +36,7 @@
         target_arch = "sparc64",
         target_arch = "x86",
         target_arch = "x86_64",
+        target_arch = "wasm32",
         target_arch = "xtensa",
     ))]
     pub type c_char = c_schar;
@@ -1323,95 +1324,95 @@
 #[path = "powerpc64/xdp.rs"]
 pub mod xdp;
 #[cfg(feature = "auxvec")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/auxvec.rs"]
 pub mod auxvec;
 #[cfg(feature = "bootparam")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/bootparam.rs"]
 pub mod bootparam;
 #[cfg(feature = "btrfs")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/btrfs.rs"]
 pub mod btrfs;
 #[cfg(feature = "elf_uapi")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/elf_uapi.rs"]
 pub mod elf_uapi;
 #[cfg(feature = "errno")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/errno.rs"]
 pub mod errno;
 #[cfg(feature = "general")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/general.rs"]
 pub mod general;
 #[cfg(feature = "if_arp")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/if_arp.rs"]
 pub mod if_arp;
 #[cfg(feature = "if_ether")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/if_ether.rs"]
 pub mod if_ether;
 #[cfg(feature = "if_packet")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/if_packet.rs"]
 pub mod if_packet;
 #[cfg(feature = "if_tun")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/if_tun.rs"]
 pub mod if_tun;
 #[cfg(feature = "image")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/image.rs"]
 pub mod image;
 #[cfg(feature = "io_uring")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/io_uring.rs"]
 pub mod io_uring;
 #[cfg(feature = "ioctl")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/ioctl.rs"]
 pub mod ioctl;
 #[cfg(feature = "landlock")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/landlock.rs"]
 pub mod landlock;
 #[cfg(feature = "loop_device")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/loop_device.rs"]
 pub mod loop_device;
 #[cfg(feature = "mempolicy")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/mempolicy.rs"]
 pub mod mempolicy;
 #[cfg(feature = "net")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/net.rs"]
 pub mod net;
 #[cfg(feature = "netlink")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/netlink.rs"]
 pub mod netlink;
 #[cfg(feature = "prctl")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/prctl.rs"]
 pub mod prctl;
 #[cfg(feature = "ptrace")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/ptrace.rs"]
 pub mod ptrace;
 #[cfg(feature = "system")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/system.rs"]
 pub mod system;
 #[cfg(feature = "vm_sockets")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/vm_sockets.rs"]
 pub mod vm_sockets;
 #[cfg(feature = "xdp")]
-#[cfg(target_arch = "riscv32")]
+#[cfg(any(target_arch = "riscv32", target_arch = "wasm32"))]
 #[path = "riscv32/xdp.rs"]
 pub mod xdp;
 #[cfg(feature = "auxvec")]
